elasticsearch_dsl/field.py

```diff
@@ -114,17 +118,30 @@ class Object(Field):
     name = 'object'
     _coerce = True
 
-    def __init__(self, doc_class=None, **kwargs):
-        self._doc_class = doc_class
+    def __init__(self, doc_class=None, dynamic=None, properties=None, **kwargs):
+        """
+        :param document.InnerDoc doc_class: base doc class that handles mapping.
+            If no `doc_class` is provided, new instance of `InnerDoc` will be created,
+            populated with `properties` and used
+        :param dynamic: whether new properties may be created dynamically.
+            Valid values are `True`, `False`, `'strict'`.
+            See https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic.html
+            for more details
+        :param dict properties: used to construct underlying mapping if no `doc_class` is provided
+        """
+        if properties is None:
+            properties = {}
         if doc_class is None:
             # FIXME import
             from .document import InnerDoc
             # no InnerDoc subclass, creating one instead...
             self._doc_class = type('InnerDoc', (InnerDoc, ), {})
-            for name, field in iteritems(kwargs.pop('properties', {})):
+            for name, field in iteritems(properties):
                 self._doc_class._doc_type.mapping.field(name, field)
-            if 'dynamic' in kwargs:
-                self._doc_class._doc_type.mapping.meta('dynamic', kwargs.pop('dynamic'))
+            if dynamic:
```

Comment:
I believe this needs to be `if dynamic is not None` as `False` is a valid value that we have to propagate through.